BSP/STM32F413/Src/BSP_SPI.c

```diff
@@ -87,15 +87,15 @@ static inline void SPI_WaitTx(spi_port_t port){
 /** SPI_WriteRead
  * @brief   Sends and receives a byte of data on the SPI line.
  * @param   txData single byte that will be sent to the device.
- * @return  rxData single byte that was read from the device.
+ * @return  rxData single byte that was read from the device. Returns 0xFF00 on error
  */
-static uint8_t SPI_WriteRead(spi_port_t port, uint8_t txData){
-    if(port >= NUM_SPI_BUSSES) return -1;
-	
+static uint16_t SPI_WriteRead(spi_port_t port, uint8_t txData){
+    if(port >= NUM_SPI_BUSSES) return 0xFF00; //set first 8 bits to 1 to signal error
+
 	SPI_Wait(port);
 	SPI_BUSSES[port]->DR = txData & 0x00FF;
 	SPI_Wait(port);
-	return SPI_BUSSES[port]->DR & 0x00FF;
+	return SPI_BUSSES[port]->DR & 0x00FF; //normal return will have the first 8 bits as 0
```

Comment:
example implementation using enum instead of magic number:
```
enum spi_status_t 
        SPI_OK = 0,
        SPI_ERROR = -1 // or 0xFFFF
        SPI_other_status = 5 // or some other number

uint16_t return_status;
if error:
        return_status = SPI_ERROR
elif valid read but return status not ok:
        return_status = SPI_other_status
else // return status ok
        return_status = SPI_OK

return spi->data | return_status
```

Reply:
I think `#define` would work better since we do not have a different error handling scheme for SPI_ERROR vs SPI_other_status. If you did want to do an enum, then I would say something like this:
```
enum spi_status_t
    SPI_OK = 0x00FF
    SPI_ERROR = 0xFF00

...blah 

return spi->data && return_status
```

Reply:
```spi->data && return_status``` would result in ```true``` most of the time hehe

Reply:
I'm a big fan of enums when you can use it though, because it has a defined type unlike ```#define```